src/index.js

```diff
@@ -544,7 +544,7 @@ function Component(props, context, opts) {
 		newComponentHook.call(this, props, context);
 	}
 }
-extend(Component.prototype = new PreactComponent(), {
+extend(Component.prototype = PreactComponent.prototype, {
```

Comment:
Ahh, I understand now. We should be able to simply extend as you noted in your comment then.